modules/streams/src/main/scala/dev/profunktor/redis4cats/pubsub/PubSub.scala

```diff
@@ -90,11 +90,14 @@ object PubSub {
   def mkSubscriberConnection[F[_]: ConcurrentEffect: ContextShift: Log, K, V](
       client: RedisClient,
       codec: RedisCodec[K, V]
-  ): Stream[F, SubscribeCommands[Stream[F, *], K, V]] =
-    Stream.resource(mkBlocker[F]).flatMap { blocker =>
+  ): Resource[F, SubscribeCommands[Stream[F, *], K, V]] =
+    mkBlocker[F].flatMap { blocker =>
       val (acquire, release) = acquireAndRelease[F, K, V](client, codec, blocker)
-      Stream.eval(Ref.of(Map.empty[K, Topic[F, Option[V]]])).flatMap { st =>
-        Stream.bracket(acquire)(release).map(new Subscriber(st, _, blocker))
+      for {
+        state <- Resource.liftF(Ref.of[F, Map[K, Topic[F, Option[V]]]](Map.empty))
+        conn <- Resource.make(acquire)(release)
+      } yield {
```

Comment:
ditto: you can remove the braces after `yield`.

Reply:
done